crates/vm/levm/src/vm.rs

```diff
@@ -1115,7 +1119,14 @@ impl VM {
                 }
             }
         };
-        (storage_slot, storage_slot_was_cold)
+
+        // Note: this is done because in update_account_storage the slot is searched in account storage, so
+        // if we do not add it here the slot is not found, so the original_value is set with 0. This is not entirely
+        // correct because caching a slot does not imply caching an account, but it is a patch (a TODO)
```

Comment:
Sorry, I didn't understand this description.
Do you think that what I'm writing down below is the same or the problem I'm describing is other thing?

> When updating account storage of an account that's not yet cached we need to store the StorageSlot in the account, so we end up caching the account because it is the most straightforward way of doing it.

Reply:
Tell me if I'm wrong, maybe I didn't understand the problem correctly

Reply:
Improved in [Update comment](https://github.com/lambdaclass/ethrex/pull/1440/commits/1702b0b041fc4fbcffdd999158dabc757bda75fd)